applications/StructuralMechanicsApplication/custom_utilities/large_displacement_variables.cpp

```diff
@@ -0,0 +1,328 @@
+    KRATOS_CATCH("");
+}
+
+void LargeDisplacementDifferentialVariables::CalculateAxisymmetricF(std::size_t IntegrationIndex)
+{
+    KRATOS_TRY;
+    BoundedMatrix<double, 2, 2> F2x2 = mF;
+    MatrixVectorUtils::InitializeMatrix(mF, 3, 3, false);
+    for (unsigned i = 0; i < 2; ++i)
+    {
+        for (unsigned j = 0; j < 2; ++j)
+            mF(i, j) = F2x2(i, j);
+        mF(i, 2) = mF(2, i) = 0.0;
+    }
+    Vector N = row(mrGeom.ShapeFunctionsValues(), IntegrationIndex);
+    const double current_radius =
+        StructuralMechanicsMathUtilities::CalculateRadius(N, mrGeom, Current);
+    const double initial_radius =
+        StructuralMechanicsMathUtilities::CalculateRadius(N, mrGeom, Initial);
+    mF(2, 2) = current_radius / initial_radius;
+    KRATOS_CATCH("");
+}
+
+LargeDisplacementKinematicVariables::LargeDisplacementKinematicVariables(
+    LargeDisplacementDifferentialVariables& rDiffVars, bool IsAxisymmetric)
+    : mrDiffVars(rDiffVars), mIsAxisymmetric(IsAxisymmetric)
+{
+}
+
+const Matrix& LargeDisplacementKinematicVariables::StrainTensor(std::size_t IntegrationIndex)
+{
+    KRATOS_TRY;
+    if (IntegrationIndex != mCurrentIntegrationIndex)
+        Recalculate(IntegrationIndex);
+    return mStrainTensor;
+    KRATOS_CATCH("");
+}
+
+Vector& LargeDisplacementKinematicVariables::StrainVector(std::size_t IntegrationIndex)
+{
+    KRATOS_TRY;
+    if (IntegrationIndex != mStrainVectorIntegrationIndex)
+    {
+        mStrainVector =
+            MathUtils<double>::StrainTensorToVector(StrainTensor(IntegrationIndex));
+        mStrainVectorIntegrationIndex = IntegrationIndex;
+    }
+    return mStrainVector;
+    KRATOS_CATCH("");
+}
+
+const Matrix& LargeDisplacementKinematicVariables::B(std::size_t IntegrationIndex)
+{
+    KRATOS_TRY;
+    if (IntegrationIndex != mCurrentIntegrationIndex)
+        Recalculate(IntegrationIndex);
+    return mB;
+    KRATOS_CATCH("");
+}
+
+void LargeDisplacementKinematicVariables::Recalculate(std::size_t IntegrationIndex)
+{
+    KRATOS_TRY;
+    CalculateStrainTensor(IntegrationIndex);
+    CalculateB(IntegrationIndex);
+    mCurrentIntegrationIndex = IntegrationIndex;
+    KRATOS_CATCH("");
+}
+
+void LargeDisplacementKinematicVariables::CalculateStrainTensor(std::size_t IntegrationIndex)
+{
+    KRATOS_TRY;
+    const Matrix& rF = mrDiffVars.F(IntegrationIndex, mIsAxisymmetric);
+    MatrixVectorUtils::InitializeMatrix(mStrainTensor, rF.size2(), rF.size2(), false);
+    noalias(mStrainTensor) = 0.5 * prod(trans(rF), rF);
+    for (std::size_t i = 0; i < mStrainTensor.size1(); ++i)
+        mStrainTensor(i, i) -= 0.5;
+    KRATOS_CATCH("");
+}
+
+void LargeDisplacementKinematicVariables::CalculateB(std::size_t IntegrationIndex)
+{
+    KRATOS_TRY;
+    if (mIsAxisymmetric)
+        CalculateB_Axisymmetric(IntegrationIndex);
+    else if (mrDiffVars.GetGeometry().WorkingSpaceDimension() == 2)
+        CalculateB_2D(mrDiffVars.F(IntegrationIndex),
```

Comment:
You used a different name elsewhere for this very same function

Reply:
I searched but can't find where you are referring to. Can you point me to it?